compio-driver/src/poll/mod.rs

```diff
@@ -338,18 +353,59 @@ fn entry_cancelled(user_data: usize) -> Entry {
     )
 }
 
+struct Notifier {
+    notify_reader: os_pipe::PipeReader,
+    notify_writer: os_pipe::PipeWriter,
+}
+
+impl Notifier {
+    pub fn new() -> io::Result<Self> {
+        let (notify_reader, notify_writer) = os_pipe::pipe()?;
+
+        // Set the reader as nonblocking.
+        let fd = notify_reader.as_raw_fd();
+        let current_flags = syscall!(libc::fcntl(fd, libc::F_GETFL))?;
```

Comment:
The implementation uses `O_CLOEXEC` as the flag on unix. Maybe we can assume it?

Reply:
OK

Reply:
No, tested and I found that the fd is not with `O_CLOEXEC`. Wierd, but revert the change now.